### PR TITLE
Feat/burn support

### PR DESCRIPTION
**Pull request by angarita-dev**
## Changelog

- Added burn support
- Style fixes

### Type of changes included:

- [x] Components
- [ ] Business Logic
- [ ] Bug fixes
- [x] Styling
- [ ] Code Refactor

### Clubhouse Tickets Related:

- Ticket 1
- Ticket 2
- Ticket 3

### Screenshots/Gifs:

*Place you changes screenshots here*

### Notes:

*Place special notes here*

### Tested on:

- [x] Chrome
- [ ] Firefox
- [ ] Safari

### CODE REVIEW

File: source/Background/Controller.js [hunk 1]
```diff
@@ -474,4 +474,86 @@ backgroundController.exposeController(
...
+  }),
+);
+
+backgroundController.exposeController(
+  'handleRequestBurnXTC',
+  async (opts, transferRequests, callId, portId) => {
+    const { callback } = opts;
+    const transfer = transferRequests?.[0];
+
+    // Answer this callback no matter if the transfer succeeds or not.
```
Comment (ChriLnth) [inline source/Background/Controller.js:529]: delete this

File: source/Background/Controller.js [hunk 2]
```diff
@@ -474,4 +474,86 @@ backgroundController.exposeController(
...
+    if (transfer?.status === 'declined') {
+      callback(null, true);
+      callback(ERRORS.TRANSACTION_REJECTED, null, [{ portId, callId }]);
+    } else {
+      const burnXTC = getKeyringHandler(HANDLER_TYPES.BURN_XTC, keyring);
+      const getBalance = getKeyringHandler(HANDLER_TYPES.GET_BALANCE, keyring);
+      const assets = await getBalance();
+
+      if (assets?.[1]?.amount * CYCLES_PER_TC > transfer.amount) {
```
Comment (ChriLnth) [inline source/Background/Controller.js:538]: this [1] is kinda monkaW, but i assume its XTC

File: source/Background/utils.js [hunk 3]
```diff
@@ -1,15 +1,36 @@
-import { validatePrincipalId, validateAccountId } from '@shared/utils/ids';
+import { validateCanisterId, validatePrincipalId, validateAccountId } from '@shared/utils/ids';
 
 import ERRORS from './errors';
 
+const validateAmount = (amount) => {
+  if (Number.isNaN(amount) || !Number.isInteger(amount) || amount < 0) {
+    return 'Invalid amount. The amount must be a positive integer. \n';
```
Comment (ChriLnth) [inline source/Background/utils.js:7]: maybe make this fn return true/false and then when u ask for it return the message 

`const validateAmount = (amount) => Number.isNaN(amount) || !Number.isInteger(amount) || amount < 0`

File: source/Background/utils.js [hunk 4]
```diff
@@ -1,15 +1,36 @@
...
 export const validateTransferArgs = ({ to, amount }) => {
   let message = null;
+  message = validateAmount(amount);
+  /*
   if (Number.isNaN(amount) || !Number.isInteger(amount) || amount < 0) {
     message = 'Invalid amount. The amount must be a positive integer. \n';
   }
+  */
   if (!validatePrincipalId(to) && !validateAccountId(to)) {
     message = 'Invalid to address. The address must be a principal Id or an account Id';
```
Comment (ChriLnth) [inline source/Background/utils.js:23]: oh i see now why u did that, hmm

**Review by ChriLnth — APPROVED** [review]
LGTM

File: source/Background/Controller.js (hunk 1 above)
Comment (angarita-dev) [reply source/Background/Controller.js:529]: The commented line?

File: source/Background/utils.js (hunk 3 above)
Comment (angarita-dev) [reply source/Background/utils.js]: Changed this to keep the function only return `true` of `false` sounds like a better implementation

File: source/Background/Controller.js (hunk 2 above)
Comment (angarita-dev) [reply source/Background/Controller.js:538]: Yep, it refers to XTC but we should probably refactor it

Comment (rocky-fleek) [reply source/Background/Controller.js:538]: you guys should've changed this :topkek:

**Pull request merged**